src/components/Profile/UserInfo/EditProfile.js

```diff
@@ -1,25 +1,39 @@
-import Input from "@mui/material/Input";
 import FormControl from "@mui/material/FormControl";
 import FormControlLabel from "@mui/material/FormControlLabel";
 import Checkbox from "@mui/material/Checkbox";
-import TextareaAutosize from "@mui/material/TextareaAutosize";
+import TextField from "@mui/material/TextField";
 import Button from "@mui/material/Button";
 import Modal from "@mui/material/Modal";
 import Box from "@mui/material/Box";
-import React from "react";
-import users from "../../TestData/users.json";
-import "../../Reused/reused.css"
+import React, {useState} from "react";
+import "../../Reused/reused.css";
+import Typography from "@mui/material/Typography";
+import {useDispatch} from "react-redux";
+import {
+  createOrUpdateUserThunk, makeUserModeratorThunk,
+  makeUserNotModeratorThunk
+} from "../../../services/user-thunks";
 
 const EditProfile = ({user}) => {
-  const userData = users[user.email];
-
-  const [open, setOpen] = React.useState(false);
-  const [admin, toggleAdmin] = React.useState(userData.admin);
+  const [open, setOpen] = useState(false);
+  const dispatch = useDispatch();
 
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
-  const handleToggleAdmin = (event) => {
-    toggleAdmin(event.target.checked);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newUserData = new FormData(event.currentTarget);
+    dispatch(createOrUpdateUserThunk({
+      ...user,
+      bio: newUserData.get("bio")
+    }));
+    console.log(newUserData.get("admin"));
+    if (newUserData.get("admin") === "admin") {
+      dispatch(makeUserModeratorThunk({userId: user.id, role: "admin"}));
```

Comment:
Did we decide the role we are giving them is type "admin"? This doesn't really matter just being picky

Reply:
This can work, maybe add this to TODOs